File: ngraph/frontend/onnx_import/src/exceptions.cpp
// Copyright (C) 2018-2021 Intel Corporation
// SPDX-License-Identifier: Apache-2.0
//

#include <sstream>

#include "exceptions.hpp"

namespace ngraph
{
    namespace onnx_import
    {
        namespace error
        {
            namespace detail
            {
                std::string get_error_msg_prefix(const Node& node)
                {
                    std::stringstream ss;
                    ss << "While validating ONNX node '" << node << "'";
                    return ss.str();
                }
            } // namespace detail
        }     // namespace error
    }         // namespace onnx_import
} // namespace ngraph

File: ngraph/python/src/ngraph/exceptions.py
# Copyright (C) 2018-2021 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

"""ngraph exceptions hierarchy. All exceptions are descendants of NgraphError."""

## @defgroup ngraph_python_exceptions Exceptions
# ngraph exceptions hierarchy. All exceptions are descendants of NgraphError.
# @ingroup ngraph_python_api


## @ingroup ngraph_python_exceptions
class NgraphError(Exception):
    """Base class for Ngraph exceptions."""


## @ingroup ngraph_python_exceptions
class UserInputError(NgraphError):
    """User provided unexpected input."""


## @ingroup ngraph_python_exceptions
class NgraphTypeError(NgraphError, TypeError):
    """Type mismatch error."""
